web/src/components/apps/AppVersionHistory.tsx

```diff
@@ -492,89 +492,6 @@ class AppVersionHistory extends Component<Props, State> {
     });
   };
 
-  getVersionDiffSummary = (version: Version) => {
-    if (!version.diffSummary || version.diffSummary === "") {
-      return null;
-    }
-    try {
-      return JSON.parse(version.diffSummary);
-    } catch (err) {
-      throw err;
-    }
-  };
-
-  renderDiff = (version: Version) => {
-    const { app } = this.props;
-    const downstream = app?.downstream;
-    const diffSummary = this.getVersionDiffSummary(version);
-    const hasDiffSummaryError =
-      version.diffSummaryError && version.diffSummaryError.length > 0;
-    let previousSequence = 0;
-    for (const v of this.state.versionHistory as Version[]) {
-      if (v.status === "pending_download") {
-        continue;
-      }
-      if (v.parentSequence < version.parentSequence) {
-        previousSequence = v.parentSequence;
-        break;
-      }
-    }
-
-    if (hasDiffSummaryError) {
-      return (
-        <div className="flex flex1 alignItems--center">
-          <span className="u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-textColor--bodyCopy">
-            Unable to generate diff{" "}
-            <span
-              className="link"
-              onClick={() => this.toggleDiffErrModal(version)}
-            >
-              Why?
-            </span>
-          </span>
-        </div>
-      );
-    } else if (diffSummary) {
-      return (
-        <div className="u-fontSize--small u-fontWeight--medium u-lineHeight--normal">
-          {diffSummary.filesChanged > 0 ? (
-            <div className="DiffSummary u-marginRight--10">
-              <span className="files">
-                {diffSummary.filesChanged} files changed{" "}
-              </span>
-              {!this.props.isHelmManaged && !downstream.gitops?.isConnected && (
-                <span
-                  className="u-fontSize--small link u-marginLeft--5"
-                  onClick={() =>
-                    this.setState({
-                      showDiffOverlay: true,
-                      firstSequence: previousSequence,
-                      secondSequence: version.parentSequence,
-                    })
-                  }
-                >
-                  View diff
-                </span>
-              )}
-            </div>
-          ) : (
-            <div className="DiffSummary">
-              <span className="files">
-                No changes to show.{" "}
-                <span
-                  className="link"
-                  onClick={() => this.toggleNoChangesModal(version)}
-                >
-                  Why?
-                </span>
-              </span>
-            </div>
-          )}
-        </div>
-      );
-    }
-  };
-
```

Comment:
these functions were only used for rendering the view diff button. moved them into the new `ViewDiffButton` component 
<img width="809" alt="Screenshot 2023-02-10 at 14 18 51" src="https://user-images.githubusercontent.com/4998130/218189710-9c1e2f81-b2f8-40f2-be5f-0c8adb2d7660.png">